File: android/app/src/main/java/ae/findmybay/data/api/dto/PaymentDtos.kt
package ae.findmybay.data.api.dto

import kotlinx.serialization.Serializable

@Serializable
data class PaymentIntentDto(
    val paymentRef: String,
    val hostedPageUrl: String,
    val processor: String,
    val amountAed: Int,
    val status: String,
)

@Serializable
data class PaymentStatusDto(
    val paymentRef: String,
    val status: String, // pending | succeeded | failed | cancelled
    val bookingId: String,
    val bookingStatus: String,
)

File: ios/FindMyBay/Data/DTO/PaymentDTOs.swift
// PaymentDTOs.swift
//
// Wire-format types for the payments/* and bookings/{id}/pay endpoints.
// Source: PaymentDtos.kt.

import Foundation

struct PaymentIntentDTO: Codable {
    let paymentRef: String
    let hostedPageUrl: String
    let processor: String
    let amountAed: Int
    let status: String
}

struct PaymentStatusDTO: Codable {
    let paymentRef: String
    /// `pending` | `succeeded` | `failed` | `cancelled`
    let status: String
    let bookingId: String
    let bookingStatus: String
}
